app/configurator/config-form.tsx

```diff
@@ -364,14 +366,102 @@ export const useSingleFilterField = ({
   };
 };
 
+export const isMultiFilterFieldChecked = (
+  chartConfig: ChartConfig,
+  dimensionIri: string,
+  value: string
+) => {
+  const filter = chartConfig.filters[dimensionIri];
+  const fieldChecked =
+    filter?.type === "multi" ? filter.values?.[value] ?? false : false;
+  return fieldChecked;
+};
+
+type CheckActionType = "SET" | "ADD";
+
+export type MultiFilterSelectionState =
+  | "SOME_SELECTED"
+  | "NONE_SELECTED"
+  | "ALL_SELECTED";
+
+const MultiFilterContext = React.createContext({
+  isFilterActive: new Set() as Set<string>,
+  checkAction: "SET" as CheckActionType, // TODO fix, use imported type
+  allValues: [] as string[],
+  selectionState: "NONE_SELECTED" as MultiFilterSelectionState,
+});
+
+export const useMultiFilterContext = () => {
+  return useContext(MultiFilterContext);
+};
+
+export const MultiFilterContextProvider = ({
+  dimensionData,
+  children,
+}: {
+  dimensionData: NonNullable<
+    DimensionValuesQuery["dataCubeByIri"]
+  >["dimensionByIri"];
+  children: React.ReactNode;
+}) => {
+  const [state] = useConfiguratorState();
+
+  const activeFilter = dimensionData
+    ? getFilterValue(state, dimensionData.iri)
+    : null;
+
+  const isFilterActive: Set<string> = useMemo(() => {
+    if (!dimensionData) {
+      return new Set();
+    }
+    const activeKeys = activeFilter
+      ? activeFilter.type === "single"
+        ? [String(activeFilter.value)]
+        : activeFilter.type === "multi"
+        ? Object.keys(activeFilter.values)
+        : []
+      : [];
+    return new Set(activeKeys);
+  }, [dimensionData, activeFilter]);
+
+  const selectionState: MultiFilterSelectionState = !activeFilter
+    ? "ALL_SELECTED"
+    : isFilterActive.size === 0
+    ? "NONE_SELECTED"
+    : "SOME_SELECTED";
+
+  const checkAction =
+    selectionState === "NONE_SELECTED" ? "SET" : ("ADD" as CheckActionType);
```

Comment:
nit: instead of using `"ADD" as CheckActionType`, use `const checkAction: CheckActionType = ...` (if the annotation is even needed?)